### PR TITLE
Preferences Screen UI Changes

### PR DESCRIPTION
**Pull request by Redeem-Grimm-Satoshi**
@wbrunette  Pref Screen UI Changes, Issue: #400

Screenshot_20230211_133450

### CODE REVIEW

**Comment by Redeem-Grimm-Satoshi** [comment]
Color, will be updated when I'll confirm with @Cveman1

**Comment by Cveman1** [comment]
Awesome work @Redeem-Grimm-Satoshi 

Please, the blue across the header is not a part of the design.

Let's fix a time tomorrow to discuss the colours?

**Comment by Redeem-Grimm-Satoshi** [comment]
@Cveman1 Yeah It's not part of the design that's why I said I'll confirm with you first so we decide on the colors after this PR is merged

**Comment by Redeem-Grimm-Satoshi** [comment]
@Cveman1 Yeah we can fix time and discuss about the colors

**Comment by Cveman1** [comment]
It's not part, you can remove it. 

How about 6pm tomorrow?

**Comment by Redeem-Grimm-Satoshi** [comment]
Check my second comment

**Comment by Redeem-Grimm-Satoshi** [comment]
Alright 6pm,

**Comment by Redeem-Grimm-Satoshi** [comment]
@Cveman1 when you talk of tomorrow, you mean today Monday right?

**Comment by Cveman1** [comment]
Yes, today. I've sent an invite

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Thu Oct 29 13:37:47 PDT 2020
+#Sat Jan 28 22:11:42 WAT 2023
 distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
```
Comment (wbrunette) [inline gradle/wrapper/gradle-wrapper.properties:3]: you should revert your code changes. the only real change is from getting "all" to "binary" which "all" is better for debugging. There is also some reordering of lines

File: services_app/build.gradle [hunk 2]
```diff
@@ -30,10 +30,11 @@ android {
         applicationId(groupId + servicesName)
         minSdkVersion(minVersion)
         targetSdkVersion(targetVersion)
-        versionCode(releaseVersionCode)
```
Comment (wbrunette) [inline services_app/build.gradle:33]: again you should revert this change since you are changing the code to not match the style that exists

File: services_app/build.gradle [hunk 3]
```diff
@@ -30,10 +30,11 @@ android {
...
+        versionCode releaseVersionCode
         versionName(versionCodeName)
         testInstrumentationRunner(instrumentationRunner)
-		android.defaultConfig.vectorDrawables.useSupportLibrary = true
+        signingConfig signingConfigs.debug
```
Comment (wbrunette) [inline services_app/build.gradle:36]: why do you need this line? It seems like something gradle should automatically assign that when you specify to gradle to either build with debug or release? It seems like you are hard-coding in something we may not want when releasing.

File: services_app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -112,7 +112,7 @@
         <activity
             android:name=".MainActivity"
             android:label="@string/app_name"
-            android:theme="@style/AppTheme">
+            android:theme="@style/Opendatakit">
```
Comment (wbrunette) [inline services_app/src/main/AndroidManifest.xml:115]: it would probably be better to rename this style to ODKX (and yes that will require a change to androidlibrary)

File: services_app/src/main/java/org/opendatakit/services/preferences/activities/AppPropertiesActivity.java [hunk 5]
```diff
@@ -101,13 +117,15 @@ protected void onCreate(Bundle savedInstanceState) {
...
     preferenceViewModel.getAdminMode().observe(this, new Observer<Boolean>() {
+
       @Override
       public void onChanged(Boolean adminMode) {
         int titleResId = adminMode ?
                 R.string.action_bar_general_settings_admin_mode :
                 R.string.action_bar_general_settings;
 
-        getSupportActionBar().setTitle(getString(titleResId, getAppName()));
+       // getSupportActionBar().setTitle(getString(titleResId, getAppName())); --> Not in use since we're dealing with custom appbar
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/preferences/activities/AppPropertiesActivity.java:127]: this code should be removed if no longer in use instead of being commented out

File: services_app/src/main/java/org/opendatakit/services/preferences/activities/ClearAppPropertiesActivity.java [hunk 6]
```diff
@@ -46,8 +50,20 @@ public class ClearAppPropertiesActivity extends Activity {
...
+      /**
+       * New dialog styling
+       * MaterialAlertDialogBuilder is standard for all ODK-X Apps
+       * OdkAlertDialogStyle present in AndroidLibrary is used to style this dialog
+       * @params uncomment and comment the necessary codes if any error do occurs!
+       */
+
+      //TODO: Make Dialog her completely material design
+
+      // AlertDialog.Builder builder = new AlertDialog.Builder(this); //--> uncomment this if any error occurs
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/preferences/activities/ClearAppPropertiesActivity.java:62]: this code should be removed if no longer in use instead of being commented out

File: services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java [hunk 7]
```diff
@@ -535,7 +541,11 @@ public PropertiesSingleton getProps() {
     }
 
     public static void showAuthenticationErrorDialog(final Activity activity, String message) {
-        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+
+        /**
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java:545]: this code should be removed if no longer in use instead of being commented out

File: services_app/src/main/res/layout-land/fragment_main.xml [hunk 8]
```diff
@@ -140,4 +140,4 @@
         app:layout_constraintTop_toBottomOf="@+id/tvLastSyncTimeLabelMain"
         tools:ignore="TextContrastCheck" />
 
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout-land/fragment_main.xml:143]: same issue, line return at end of file expected

File: services_app/src/main/res/layout-land/fragment_update_server_settings.xml [hunk 9]
```diff
@@ -150,4 +138,4 @@
 
     </androidx.constraintlayout.widget.ConstraintLayout>
 
-</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout-land/fragment_update_server_settings.xml:153]: same issue, line return at end of file expected

File: services_app/src/main/res/layout-land/sync_activity.xml [hunk 10]
```diff
@@ -67,4 +68,4 @@
         android:layout_width="wrap_content"
         android:layout_height="match_parent" />
 
-</androidx.drawerlayout.widget.DrawerLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout-land/sync_activity.xml:70]: same issue, line return at end of file expected

File: services_app/src/main/res/layout-land/sync_launch_fragment.xml [hunk 11]
```diff
@@ -215,4 +215,4 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/btnSignInSync" />
```
Comment (wbrunette) [inline services_app/src/main/res/layout-land/sync_launch_fragment.xml:217]: same issue, line return at end of file expected

File: services_app/src/main/res/layout-land/sync_launch_fragment.xml [hunk 12]
```diff
@@ -215,4 +215,4 @@
...
 
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout-land/sync_launch_fragment.xml:218]: same issue, line return at end of file expected

File: services_app/src/main/res/layout-land/verify_server_settings_launch_fragment.xml [hunk 13]
```diff
@@ -216,4 +216,4 @@
         app:layout_constraintTop_toBottomOf="@+id/tvLastSyncTimeVerify"
         tools:ignore="TextContrastCheck" />
 
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout-land/verify_server_settings_launch_fragment.xml:219]: same issue, line return at end of file expected

File: services_app/src/main/res/layout-v21/drawer_header.xml [hunk 14]
```diff
@@ -39,4 +39,4 @@
 
     </com.google.android.material.appbar.AppBarLayout>
 
-</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout-v21/drawer_header.xml:42]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/activity_app_properties.xml [hunk 15]
```diff
@@ -1,8 +1,45 @@
...
+    </androidx.core.widget.NestedScrollView>
+
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:orientation="vertical"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:layout_marginTop="55dp"
     android:id="@+id/app_properties_content">
-
-</FrameLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/activity_app_properties.xml:8]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/checkpoint_resolver_chooser_list.xml [hunk 16]
```diff
@@ -72,4 +76,4 @@ the License.
         android:textColor="@color/black"
         android:textSize="21sp" />
 
-</LinearLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/checkpoint_resolver_chooser_list.xml:75]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/checkpoint_resolver_field_list.xml [hunk 17]
```diff
@@ -51,4 +58,4 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent" />
 
-</LinearLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/checkpoint_resolver_field_list.xml:54]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/conflict_resolver_chooser_list.xml [hunk 18]
```diff
@@ -72,4 +76,4 @@ the License.
         android:textColor="@color/black"
         android:textSize="21sp" />
 
-</LinearLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/conflict_resolver_chooser_list.xml:75]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/conflict_resolver_field_list.xml [hunk 19]
```diff
@@ -51,4 +57,4 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent" />
 
-</LinearLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/conflict_resolver_field_list.xml:54]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/drawer_header.xml [hunk 20]
```diff
@@ -38,4 +38,4 @@
 
     </com.google.android.material.appbar.AppBarLayout>
 
-</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/drawer_header.xml:41]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/dropdown_list_item.xml [hunk 21]
```diff
@@ -3,4 +3,4 @@
     style="@style/SmallTextViewStyle"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:padding="@dimen/list_item_padding" />
```
Comment (wbrunette) [inline services_app/src/main/res/layout/dropdown_list_item.xml:6]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/fragment_choose_sign_in_type.xml [hunk 22]
```diff
@@ -68,4 +68,4 @@
         app:layout_constraintTop_toBottomOf="@+id/btnAnonymousSignInLogin"
         tools:ignore="TextContrastCheck" />
 
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/fragment_choose_sign_in_type.xml:71]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/fragment_main.xml [hunk 23]
```diff
@@ -144,13 +144,12 @@
...
+        android:layout_height="50dp"
         android:layout_margin="@dimen/portrait_regular_margin"
         android:text="@string/drawer_sign_in_button_text"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/tvLastSyncTimeMain"
         tools:ignore="TextContrastCheck" />
-
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/fragment_main.xml:156]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/fragment_set_credentials.xml [hunk 24]
```diff
@@ -100,4 +100,4 @@
         app:layout_constraintTop_toBottomOf="@+id/inputPasswordLogin"
         tools:ignore="TextContrastCheck" />
 
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/fragment_set_credentials.xml:103]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/fragment_update_server_settings.xml [hunk 25]
```diff
@@ -149,4 +138,4 @@
 
     </androidx.constraintlayout.widget.ConstraintLayout>
 
-</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/fragment_update_server_settings.xml:152]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/list_item_concordant_row.xml [hunk 26]
```diff
@@ -48,4 +48,4 @@
 
         </RelativeLayout>
         
-</LinearLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/list_item_concordant_row.xml:51]: same issue, line return at end of file expected

File: services_app/src/main/res/layout/list_item_conflict_row.xml [hunk 27]
```diff
@@ -106,4 +106,4 @@
 	    
 	</RelativeLayout>
 
-</LinearLayout>
```
Comment (wbrunette) [inline services_app/src/main/res/layout/list_item_conflict_row.xml:109]: same issue, line return at end of file expected

File: services_app/src/main/res/values/strings.xml [hunk 28]
```diff
@@ -258,7 +258,7 @@
     <string name="admin_password_disabled_click_to_set">No Admin password. Click to enable.</string>
     <string name="admin_password_settings">Admin Password</string>
     <string name="admin_password">Admin Password</string>
-    <string name="change_admin_password">Change admin password</string>
+    <string name="change_admin_password">Privacy</string>
```
Comment (wbrunette) [inline services_app/src/main/res/values/strings.xml:261]: Admin password is not privacy ... seems like a change we should discuss at meeting

**Review by wbrunette — CHANGES_REQUESTED** [review]

**Comment by Redeem-Grimm-Satoshi** [comment]
@wbrunette thanks for spotting out the changes, working on them..

File: services_app/src/main/java/org/opendatakit/services/utilities/TableHealthValidator.java [hunk 29]
```diff
@@ -135,12 +146,19 @@ public void onClick(DialogInterface dialog, int id) {
         displayActivity.startActivity(i);
       }
     });
-    AlertDialog dialog = builder.create();
+    androidx.appcompat.app.AlertDialog dialog = builder.create();
```
[the remark this answers is not in the document]
Comment (Redeem-Grimm-Satoshi) [reply services_app/src/main/java/org/opendatakit/services/utilities/TableHealthValidator.java:149]: @wbrunette Please how do you mean? Didn't quite understand "fully qualified path"

**Comment by Redeem-Grimm-Satoshi** [comment]
@wbrunette I've added some commits to resolve the issues you spotted out

File: services_app/src/main/res/values/strings.xml (hunk 28 above)
Comment (wbrunette) [reply services_app/src/main/res/values/strings.xml:261]: applied label update to wrong string

**Comment by Redeem-Grimm-Satoshi** [comment]
@wbrunette Just added some new commits to resolve the current issues you listed

File: services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java [hunk 30]
```diff
@@ -428,8 +429,13 @@ private void onFirstLaunch() {
...
 
-        androidx.appcompat.app.AlertDialog alertDialog = new androidx.appcompat.app.AlertDialog
-                .Builder(this)
+
+        /**
+         * MaterialAlertDialogBuilder is standard for all ODK-X Apps
+         * OdkAlertDialogStyle present in AndroidLibrary is used to style this dialog
+         */
+
+        androidx.appcompat.app.AlertDialog alertDialog = new MaterialAlertDialogBuilder(getApplicationContext(),R.style.OdkXAlertDialogStyle)
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java:438]: Why isn't the type simply MaterialAlertDialogBuilder instead of androidx.appcompat.app.AlertDialog?

File: services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java [hunk 31]
```diff
@@ -535,21 +541,32 @@ public PropertiesSingleton getProps() {
...
-        dialog.show();
+
+        /**
+         * New dialog styling
+         * MaterialAlertDialogBuilder is standard for all ODK-X Apps
+         * OdkAlertDialogStyle present in AndroidLibrary is used to style this dialog
+         * @params
+         */
+
+        AlertDialog alertDialog = new MaterialAlertDialogBuilder(activity,R.style.OdkXAlertDialogStyle)
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java:552]: Why isn't the type simply MaterialAlertDialogBuilder instead of androidx.appcompat.app.AlertDialog?

File: services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/ChooseSignInTypeFragment.java [hunk 32]
```diff
@@ -249,8 +252,15 @@ void perhapsEnableButtons() {
...
-                .Builder(requireActivity())
+
+        /**
+         * New dialog styling
+         * MaterialAlertDialogBuilder is standard for all ODK-X Apps
+         * OdkAlertDialogStyle present in AndroidLibrary is used to style this dialog
+         * @params change MaterialAlertDialogBuilder to AlertDialog.Builder in case of any error and remove R.style... param!
+         */
+
+        AlertDialog alertDialog = new MaterialAlertDialogBuilder(requireActivity(),R.style.OdkXAlertDialogStyle)
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/ChooseSignInTypeFragment.java:263]: Why isn't the type simply MaterialAlertDialogBuilder instead of androidx.appcompat.app.AlertDialog?

File: services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/LoginFragment.java [hunk 33]
```diff
@@ -453,8 +455,15 @@ private void performActionOnSyncComplete(SyncStatus syncStatus){
...
-              .Builder(requireActivity())
+
+      /**
+       * New dialog styling
+       * MaterialAlertDialogBuilder is standard for all ODK-X Apps
+       * OdkAlertDialogStyle present in AndroidLibrary is used to style this dialog
+       * @params change MaterialAlertDialogBuilder to AlertDialog.Builder in case of any error and remove R.style... param!
+       */
+
+      AlertDialog alertDialog = new MaterialAlertDialogBuilder(requireActivity(),R.style.OdkXAlertDialogStyle)
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/LoginFragment.java:466]: Why isn't the type simply MaterialAlertDialogBuilder instead of androidx.appcompat.app.AlertDialog?

**Review by wbrunette — CHANGES_REQUESTED** [review]

**Comment by Redeem-Grimm-Satoshi** [comment]
@wbrunette thanks. From the docs https://developer.android.com/reference/com/google/android/material/dialog/MaterialAlertDialogBuilder, The type of dialog returned is still an AlertDialog; there is no specific Material implementation of AlertDialog

File: services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java (hunk 30 above)
Comment (wbrunette) [reply services_app/src/main/java/org/opendatakit/services/sync/actions/activities/AbsSyncBaseActivity.java:438]: UPDATED: why isn't the type "AlertDialog"? Why the fully qualified " androidx.appcompat.app.AlertDialog"

File: services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java [hunk 34]
```diff
@@ -729,8 +737,15 @@ private void performActionOnSyncComplete(SyncStatus syncStatus){
...
-            .Builder(requireActivity())
+
+    /**
+     * New dialog styling
+     * MaterialAlertDialogBuilder is standard for all ODK-X Apps
+     * OdkAlertDialogStyle present in AndroidLibrary is used to style this dialog
+     * @params change MaterialAlertDialogBuilder to AlertDialog.Builder in case of any error and remove R.style... param!
+     */
+
+    androidx.appcompat.app.AlertDialog alertDialog = new MaterialAlertDialogBuilder(requireActivity(),R.style.OdkXAlertDialogStyle)
```
Comment (wbrunette) [inline services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java:748]: why isn't the type "AlertDialog"? Why the fully qualified " androidx.appcompat.app.AlertDialog"

File: services_app/src/main/res/values/strings.xml [hunk 35]
```diff
@@ -258,8 +258,10 @@
     <string name="admin_password_disabled_click_to_set">No Admin password. Click to enable.</string>
     <string name="admin_password_settings">Admin Password</string>
     <string name="admin_password">Admin Password</string>
-    <string name="change_admin_password">Change admin password</string>
-    <string name="enable_admin_password">Enable user restrictions</string>
+    <string name="user_restrictions">Restrictions</string>
+    <string name="enable_user_restrictions">Enable user restrictions</string>
+    <string name="change_admin_password">Enable Admin Password</string>
```
Comment (wbrunette) [inline services_app/src/main/res/values/strings.xml:263]: Shouldn't this be "Change" Admin Password? The problem is two strings both have the same value. String names "change_admin_password" and "enable_admin_password" currently have the same value which seems incorrect as each string value should be unique.

File: services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java (hunk 34 above)
Comment (Redeem-Grimm-Satoshi) [reply services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java:748]: I just fixed it now

**Comment by Redeem-Grimm-Satoshi** [comment]
@wbrunette I did the minor fixes in this new commit

**Review by wbrunette — APPROVED** [review]
